br/pkg/lightning/backend/external/byte_reader.go

```diff
@@ -54,16 +65,52 @@ func openStoreReaderAndSeek(
 
 // newByteReader wraps readNBytes functionality to storageReader. It will not
 // close storageReader when meet error.
-func newByteReader(ctx context.Context, storageReader storage.ReadSeekCloser, bufSize int) (*byteReader, error) {
+func newByteReader(ctx context.Context, storageReader storage.ExternalFileReader, bufSize int, st storage.ExternalStorage, name string, defaultUseConcurrency bool) (*byteReader, error) {
+	conReader, err := newSingeFileReader(ctx, st, name, 8, ConcurrentReaderBufferSize)
+	if err != nil {
+		return nil, err
+	}
 	r := &byteReader{
 		ctx:           ctx,
 		storageReader: storageReader,
 		buf:           make([]byte, bufSize),
 		bufOffset:     0,
+		conReader:     conReader,
 	}
+	r.SwitchReaderMode(defaultUseConcurrency)
 	return r, r.reload()
 }
 
+// SwitchReaderMode switches to concurrent reader.
+func (r *byteReader) SwitchReaderMode(useConcurrent bool) {
```

Comment:
```suggestion
func (r *byteReader) switchReaderMode(useConcurrent bool) {
```

Maybe unexport this because `byteReader` is unexported.